DTS: flash: Various minor flash related cleanups

galak opened the pull request.

galak committed:
dts: bindings: add missing binding for st,stm32l0-flash-controller

The st,stm32l0-flash-controller did not have a binding, add one for it.

Also made a comment in stm32l0.dtsi that the driver doesn't currently
support this controller.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: bindings: Remove zephyr,native-posix-flash-controller binding

Nothing uses the zephyr,native-posix-flash-controller binding so
remove it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: ti: lm3s6965: Add flash controller node and binding

Add a flash controller node and place the soc-nv-flash under that node.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: stm32: H7: Fixup flash nodes

Add missing compatible and label for flash nodes on STM32 H7 SoC dtsi
files.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

erwango commented on dts/bindings/flash_controller/st,stm32l0-flash-controller.yaml:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020, Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32 L0 flash controller
```
Since STM32L0 flash driver is not supported, I'd prefer to remove the flash driver node in stm32l0.dtsi

galak replied:
Why?   The device tree should represent the hardware.  Part of the reason I'm making these clean ups is I want to require that 'soc-nv-flash' is a child of an actual flash controller.  So removing this for L0 would violate that.

erwango replied:
Ok my point is I'd like to see dts to reflect actual support in zephyr:
If a node is present, it is supported.
Since L0 flash driver is not implemented, having it in dts makes this confusing.
But if this is not possible given the 'soc-nv-flash' availability, then a comment should be (/have already been) added to make it clear.

galak replied:
Where should I add the comment?

galak replied:
Comment added.

nandojve commented on dts/arm/atmel/sam4e16c.dtsi:
```diff
@@ -7,11 +7,12 @@
-	flash0: flash@400000 {
-		reg = <0x00400000 DT_SIZE_K(1024)>;
-	};
+&flash0 {
+	reg = <0x00400000 DT_SIZE_K(1024)>;
+};
```
I did some tests with full path and works fine because of MCUboot. In general, we I saw constructions like:
```
/ {
	soc {
		flash-controller@400e0a00 {
			compatible = "atmel,sam-flash-controller";
			label = "FLASH_CTRL";
			reg = <0x400e0a00 0x200>;
			peripheral-id = <6>;

			#address-cells = <1>;
			#size-cells = <1>;

			flash0: flash@400000 {
				compatible = "soc-nv-flash";
				label = "FLASH_0";

				write-block-size = <16>;
				erase-block-size = <8192>;
			};
		};
```
It seems more clear when define the full path:
```
/ {
	soc {
		flash-controller@400e0a00 {
			flash0: flash@400000 {
				reg = <0x00400000 DT_SIZE_K(1024)>;
			};
		};
	};
};
```

galak replied:
Yeah, I was being a bit lazy and using the node label reference form so that way the location/path didn't have to be explicit.

galak replied:
Changed style back to show full SoC path.  Also moved the sram nodes while we're at it.

galak committed:
dts: atmel: sam: cleanup flash / sram nodes

Move flash node under flash controller and sram nodes under soc as it
is for most SoC dts files.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
